lib/Mojo/Log.pm

```diff
@@ -60,7 +67,12 @@ sub _default {
   return "[$time] [$$] [$level] " . join "\n", @_, '';
 }
 
-sub _log { shift->emit('message', pop, ref $_[0] eq 'CODE' ? $_[0]() : @_) }
+sub _log {
+  my ($self, $level) = (shift, pop);
+  my @msgs = ref $_[0] eq 'CODE' ? $_[0]() : @_;
+  $msgs[0] = "$self->{context} $msgs[0]" if $self->{context};
```

Comment:
Why isn't this done inside _default() and _short()? I was thinking if context() could be a getter as well, then custom log format() could format it any way they wanted.

Reply:
This has been discussed in great detail on IRC, it's unfortunate you did not participate. There is no clean way to do that without breaking backwards compatibility. We might do it in the next major release.

Reply:
Since there's log public logs from IRC, then I thought it might be worth mentioning it here as well. Anyways, I'll resolve this comment as well.